resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/rest/kubernetes/KubernetesSparkRestServer.scala

```diff
@@ -45,6 +46,7 @@ private case class KubernetesSparkRestServerArguments(
   def validate(): KubernetesSparkRestServerArguments = {
     require(host.isDefined, "Hostname not set via --hostname.")
     require(port.isDefined, "Port not set via --port")
+    require(basePath.isDefined, "Base path not defined via --base-path")
```

Comment:
I think we want the base path to be consistent for both this and the Spark UI regardless if the Ingress is used or not. That reduces the number of end results that would come up depending on the configuration. For example someone may migrate their application to submit through Ingress instead of previously submitting through NodePort and they shouldn't have to rewrite the base path for their Spark UI.